src/upload.rs

```diff
@@ -205,4 +217,72 @@ impl<'a> Uploader<'a> {
             .map_err(|e| Error::RequestError(Box::new(e)))?
             .into_string()?)
     }
+
+    /// Retrieves and prints the files on server.
+    pub fn retrieve_list(&self, prettify: bool) -> Result<()> {
+        let mut url = Url::parse(&self.config.server.address)?;
+        if !url.path().to_string().ends_with('/') {
+            url = url.join(&format!("{}/", url.path()))?;
+        }
+        url = url.join("list")?;
```

Comment:
This code looks familiar... Ah!

https://github.com/orhun/rustypaste-cli/blob/2032fcf99df40ead114656e05afb65727623880c/src/upload.rs#L193-L96

So maybe it is the right time to extract this into a function?

Reply:
Yes, a function makes sense.

Reply:
done

Reply:
Oops, I shouldn't resolve them. This is usually done by the person who started the thread. Sorry. Unresolving.